Eatery/Controllers/Eateries/Campus/ScrollableViewController.swift

```diff
@@ -0,0 +1,374 @@
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        let screenSize = UIScreen.main.bounds.height
+        let tabBarHeight = (self.tabBarController?.tabBar.frame.height) ?? 0
+        let smallOffset: CGFloat = CampusEateryExpandedMenuViewController.heightConst
+
+        // Scroll Offset is used below because it accounts for NavBar height and edge insets
+        // reducedScreenSize accounts for the height of the screen minus all nav and tab bars
+        let reducedScreenSize = screenSize + (scrollOffset - tabBarHeight) + smallOffset
+        if originalViews.count > 0 {
+            let finalView = originalViews[originalViews.count - 1]
+            let checkHeight = reducedScreenSize - (menuItemsStack.frame.maxY - finalView.frame.minY)
+            fillerSpaceHeight = checkHeight > 0 ? checkHeight : 0
+        }
+
+        self.scrollView?.contentSize.height += fillerSpaceHeight
+    }
+
+    private func createTabBarStack() {
+        headerContainer = UIView()
+        headerContainer.layer.zPosition = 3
+        headerContainer.backgroundColor = .white
+        headerContainer.directionalLayoutMargins = NSDirectionalEdgeInsets(top: 0, leading: 16, bottom: 0, trailing: 16)
+
+        headerContainer.snp.makeConstraints { make in
+            make.height.equalTo(58)
+        }
+
+        let titleLabel = UILabel()
+        titleLabel.text = "Menu"
+        titleLabel.textColor = .black
+        titleLabel.font = .boldSystemFont(ofSize: 26)
+        titleLabel.preservesSuperviewLayoutMargins = true
+        headerContainer.addSubview(titleLabel)
+
+        filterButton = ExtendedFilterButton(frame: .zero, inactiveColor: .systemGray, activeColor: .eateryBlue)
+        filterButton.addTarget(self, action: #selector(filterButtonPressed), for: .touchUpInside)
+        filterButton.preservesSuperviewLayoutMargins = true
+        headerContainer.addSubview(filterButton)
+
+        filterLabel = UILabel()
+        filterLabel.textColor = .eateryBlue
+        filterLabel.font = .boldSystemFont(ofSize: 11)
+        filterLabel.text = ""
+        filterLabel.preservesSuperviewLayoutMargins = true
+        headerContainer.addSubview(filterLabel)
+
+        separatorView = SeparatorView(insets: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0))
+        separatorView.tintColor = .clear
+        separatorView.isHidden = true
+        headerContainer.addSubview(separatorView)
+
+        titleLabel.snp.makeConstraints { make in
+            make.leading.equalToSuperview().offset(tabBarPadding)
+            make.centerY.equalToSuperview()
+            make.bottom.equalToSuperview()
+        }
+
+        filterButton.snp.makeConstraints { make in
+            make.centerY.equalTo(titleLabel.snp.centerY)
+            make.height.equalTo(titleLabel.snp.height).multipliedBy(0.3)
+            make.trailing.equalToSuperview().offset(-tabBarPadding)
+            make.width.equalTo(filterButton.snp.height).multipliedBy(1.71)
```

Comment:
0.3 and 1.71 are very specific, how'd you derive these?

also as an aside, it's nice to put all the constraints as constants somewhere in the file where devs can find it easily instead of scrolling to find where exactly the constraints were made - however, if this is how eatery has been doing constraint constants then just leave it as it is for consistency

Reply:
Based off what I've seen from other Eatery files, it seems constraints are hardcoded like they are here, rather than using variables.

I derived those by simple trial and error to try and make the button look as similar to the Figma file as possible.  However, I believe 1.71 was a typo and should be 1.7